.github/workflows/ci-baseline.yml

```diff
@@ -28,75 +28,92 @@ jobs:
       gtest_dir: /usr/local/gtest/install
-      small: small
-      medium: medium
-      large: large
 
     steps:
       - uses: actions/checkout@v3
 
-      - name: configure
+      - name: Configure
         run: |
           cmake -B builddir \
-            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.config.mode }} \
-            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.build_type }} \
             -D CMAKE_C_COMPILER=$CC \
             -D CMAKE_CXX_COMPILER=$CXX \
-            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
-            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir \
+            -D CMAKE_CXX_FLAGS='' \
+            -D CMAKE_INSTALL_PREFIX:PATH=../install \
+            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            ${{ matrix.cmake_extra_args }} \
             -D GTEST_ROOT=$gtest_dir \
             -D PRESSIO_ENABLE_DEBUG_PRINT=ON \
             -D PRESSIO_ENABLE_TESTS:BOOL=ON \
-            -D CMAKE_INSTALL_PREFIX:PATH=../install \
-            -D CMAKE_CXX_FLAGS=''
+            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
+            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir
 
-      # In this stage small functional tests are built and run
-      - name: build-test-${{ env.small }}
+      - name: Build
         run: |
-          cd builddir/tests/functional_${{ env.small }}
-          make -j $num_cpus
-          ctest -j $num_cpus --output-on-failure
+          cmake --build builddir --target install
 
-      # In this stage medium functional tests are built and run
-      - name: build-test-${{ env.medium }}
+      - name: Test
+        working-directory: builddir
         run: |
-          cd builddir/tests/functional_${{ env.medium }}
-          make -j $num_cpus
           ctest -j $num_cpus --output-on-failure
 
-      # In this stage large functional tests are built and run
-      - name: build-test-${{ env.large }}
+      - name: Example - CMake installed
+        working-directory: examples/cmake_installed
         run: |
-          cd builddir/tests/functional_${{ env.large }}
-          make -j $num_cpus
-          ctest -j $num_cpus --output-on-failure
+          cmake -B builddir \
+            -DCMAKE_PREFIX_PATH=../../install \
+          && cmake --build builddir           \
+          && ./builddir/pressio-example
+
+      - name: Example - CMake add_subdirectory
+        working-directory: examples/cmake_subdir
+        run: |
+          cmake -B builddir \
+            -DEIGEN_INCLUDE_DIR=$eigen_inc_dir \
+          && cmake --build builddir            \
+          && ./builddir/pressio-example
+
+      - name: Example - include only
+        working-directory: examples
+        run: |
+          $CXX \
+            -DPRESSIO_ENABLE_TPL_EIGEN \
+            -I$eigen_inc_dir           \
+            -I../include            \
+            -c main.cpp                \
+          && $CXX -o pressio-example main.o \
+          && ./pressio-example
```

Comment:
@fnrizzi Please have a look at this and `main.cpp`. Unless we can prepare more sophisticated examples, I don't think there's much value in creating a separate repository.

Reply:
i made this: https://github.com/Pressio/pressio-templates-usage-as-library

Reply:
Examples / templates moved to the `pressio-templates-usage-as-library`. They will be run by the `pressio` CI before running the tests.